dashboard/app/views/levels/_dialogHelper.js

```diff
@@ -35,14 +35,24 @@
     // Use our prefabricated dialog content.
     $(".modal-body").append($("#" + type + "-dialogcontent").clone(true));
 
-    // Clicking the okay button in the dialog box dismisses it.
+    // Clicking the okay button in the dialog box dismisses it, and calls the callback.
     $(".modal-body #ok-button").click(function () {
       dialog.hide();
+      callback && callback();
+    });
+
+    // Clicking the cancel button in the dialog box dismisses it.
+    $(".modal-body #cancel-button").click(function () {
+      dialog.hide();
     });
 
     dialog.show();
   }
 
+  window.showStartOverDialog = function(callback) {
```

Comment:
we have the same issue as w/ the delete account button that red, which we
intend to mean "danger" means "ooh, press me!" to some people...

On Fri, Aug 14, 2015 at 4:54 PM, Josh Lory <notifications@github.com> wrote:

> In dashboard/app/views/levels/_dialogHelper.js
> <https://github.com/code-dot-org/code-dot-org/pull/3517#discussion_r37129297>
> :
>
> >      });
> >
> >      dialog.show();
> >    }
> >
> > +  window.showStartOverDialog = function(callback) {
>
> Untracked global scope dependency.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/3517/files#r37129297>.
>